src/Integration/Telemetry/TelemetryManager.cs

```diff
@@ -67,92 +62,86 @@ public bool IsAnonymousDataShared
             {
                 throw new ArgumentNullException(nameof(telemetryClient));
             }
+            if (timerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(timerFactory));
+            }
+            if (knownUIContexts == null)
+            {
+                throw new ArgumentNullException(nameof(knownUIContexts));
+            }
 
             this.solutionBindingTracker = solutionBindingTracker;
             this.telemetryClient = telemetryClient;
             this.telemetryRepository = telemetryRepository;
-
-            string path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "msenv.dll");
-            visualStudioVersion = File.Exists(path)
-                ? FileVersionInfo.GetVersionInfo(path).ProductVersion
-                : string.Empty;
+            this.knownUIContexts = knownUIContexts;
 
             if (this.telemetryRepository.Data.InstallationDate == DateTime.MinValue)
             {
-                this.telemetryRepository.Data.InstallationDate = DateTime.Now;
+                this.telemetryRepository.Data.InstallationDate = DateTime.Now; // TODO: Use some mockable clock
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)